src/dtls_combine.rs

```diff
@@ -0,0 +1,219 @@
+use core::array;
+
+use nom::error::{make_error, ErrorKind};
+use nom::{Err, IResult};
+
+use crate::{
+    parse_dtls_message_handshake, DTLSMessage, DTLSMessageHandshake, DTLSMessageHandshakeBody,
+};
+
+const MAX_FRAGMENTS: usize = 50;
+
+/// Combine the given fragments into one. Returns true if the fragments made a complete output.
+/// The fragments are combined in such a way that the output constitutes a complete DTLSMessage.
+///
+/// Returns `None` if the fragments are not complete.
+///
+/// Errors if:
+///
+/// 1. The output is not big enough to hold the reconstituted messages
+/// 2. Fragments are not of the same type (for example ClientHello mixed with Certificate)
+/// 3. (Total) length field differs between the fragments
+/// 4. Fragment offset/length are not consistent with total length
+/// 5. The DTLSMessageHandshakeBody in the message is not a Fragment
+/// 6. The message_seq differs between the fragments.
+///
+/// Panics if there are more than 50 fragments.
```

Comment:
(minor) This comment is not correct now, it will return an error (and not panic)

Reply:
Yep will fix. Following the logic (if we keep that), this should be error case 7 or something.